### PR TITLE
ADO: Improved Setup Error Handling

### PR DESCRIPTION
**Pull request by j-dunham**
## What is changing?
The purpose of these changes are to give the users a better experience when they encounter errors during the setup script by:
- Adding the ability to skip creating a project if it already exists.  This is useful if a user encountered a error after the project was created and reran the command.
```command
@j-dunham ➜ /workspaces/gai-labs (main) $ ./azure_devops/bootstrap/setup --organization $org --project ado-lab --access-token $pat
Project 'ado-lab' already exists. Would you like to continue? [y/N]y
Extracting assets...
Creating repository ado-lab...
```
- Customizing the error message based on the ADO API status code and using `abort` to prevent the stack trace from making the error hard to read

Example of a bad token
```command
Error encountered on request:https://dev.azure.com/jd-testing-org/_apis/projects?api-version=6.0
  StatusCode:302
  Message: 'Please check your Azure DevOps access token is correctly entered.'
```

Example of bad permissions
```command
Error encountered on request:https://dev.azure.com/jd-testing-org/_apis/projects?api-version=6.0
  StatusCode:401
  Message: 'Please check your Azure DevOps access token has the correct permissions.'
```

## How was this tested
👀 

Closes github/valet#4974

### CODE REVIEW

**Comment by JenniferKerns** [comment]
Are these revisions too wordy?
- Please ensure that your Azure DevOps access token is entered correctly.
- Please ensure that your Azure DevOps access token has the correct permissions.

**Comment by j-dunham** [comment]
> Are these revisions too wordy?
> 
> * Please ensure that your Azure DevOps access token is entered correctly.
> * Please ensure that your Azure DevOps access token has the correct permissions.

I don't think so.  Looks good to me!

**Review by ethanis — APPROVED** [review]
